docs: edits to remove jargon (#43000)
angular/angular: aio/content/guide/service-worker-config.md

* `**` matches 0 or more path segments.
* `*` matches 0 or more characters excluding `/`.
* `?` matches exactly one character excluding `/`.
* The `!` prefix marks the pattern as being negative, meaning that only files that don't match the pattern are included.

Example patterns:

* `/**/*.html` specifies all HTML files.
* `/*.html` specifies only HTML files in the root.
* `!/**/*.map` exclude all sourcemaps.

> Each section of the configuration file is described in the following section.

Reviewer: This sounds a little confusing to me, because (a) we use the word "section" to refer to two different things in the same sentence and (b) there is not _one_ following section that describes the configuration file properties, but one section per property. So, maybe change it to something like:

```suggestion
The following sections describe each property of the configuration file.
```

Author: Done

### Handling navigation requests

> The ServiceWorker redirectd navigation requests that don't match any `asset` or `data` group to the specified [index file](#index-file). A request is considered to be a navigation request if:

Reviewer: ```suggestion
The ServiceWorker redirects navigation requests that don't match any `asset` or `data` group to the specified [index file](#index-file). A request is considered to be a navigation request if:
```

Author: done

1. Its [mode](https://developer.mozilla.org/en-US/docs/Web/API/Request/mode) is `navigation`.
2. It accepts a `text/html` response (as determined by the value of the `Accept` header).

> 3. Its URL matches certain criteria (see the following).

Reviewer: Is that correct? (non-native speaker question 😅)

Author: Yes

Reviewer 2: Hi @gkalpak ! For reference, here's the entry in the Google Developers Style Guide: https://developers.google.com/style/word-list?hl=en#letter-b. I'm afraid it doesn't explain why following is better, but we do try to comply with the style guide as much as possible!
